app/controllers/users_controller.rb

```diff
@@ -1,4 +1,30 @@
 class UsersController < ApplicationController
+
   def new
+    @user = User.new
   end
+
+  def show
+    @user = User.find_by( id: params[:id])
```

Comment:
đoạn code này có nghĩa gì

Reply:
dạ, đoạn này sẽ tìm kiếm 1 user vơis id được truyền tới từ url, sau đó sẽ hiển thị thông tin user trên web, 
thay vì sử dụng find thì e sẽ sử dụng find_by,  để khi k tìm thấy thì nó sẽ trả veef nil chứ k báo lỗi như find !

Reply:
theo như em nói, nếu ko tìm thấy sẽ trả về nil, nhưng nếu em đem 1 biến nil ra sử dụng ngoài view thì điều gì sẽ xảy ra

Reply:
hiện tại trên view e code là @user.name thì sẽ bị lỗi vì nó 
- fix lỗi trên thì e nghĩ là sẽ viết 1 method trong helper để check xem có nil k!. nếu là nil thì xuất thông báo user k tồn tại, nếu k nil thì hiển thị info user ạ!

Reply:
ý tưởng rất tốt, nhưng sao ko viết luôn tại controller mà phải qua helper làm gì nhỉ

Reply:
if @user.nil?
      flash[:error] = "User  not found "
      redirect_to root_url
    end
giả sử nếu viết trong controller thì viết xử lý nó luôn trong def show như thế này được không a?

Reply:
cũng là 1 ý hay